### ISSUE TITLE
onChange on input type[checkbox] not working

### ISSUE THREAD

**Issue by kushalmahajan** [styled-components/vue-styled-components#66]
This is my simple functional component in Vuejs

```
import { Label, Input, Checkmark } from "./styles";
export default {
  functional: true,
  model: {
    prop: "checked",
    event: "change"
  },
  props: {
    checked: Boolean
  },
  // eslint-disable-next-line
  render(h, { props, listeners}) {
    console.log(listeners);
    const changeHandler = listeners.change ? listeners.change : () => {};
    return (
      <Label>
        <Input
          type="checkbox"
          checked={props.checked}
          onChange={e => {
            console.log("checked", e.target.checked);
            changeHandler(e);
          }}
        />
        <Checkmark class="checkmark" />
      </Label>
    );
  }
};
```

The component simply won't console.log when used like

```

<VCheckbox
              key={this.checkSelections[idx]}
              checked={this.checkSelections[idx]}
              onChange={e => {
                this.$set(this.checkSelections, idx, e.target.checked);
              }}
/>
```
I'm using **Vuejs - 2.5.x**

If I console.log my `listeners` they definitely log change. 

Second, if I use `nativeOnChange` instead it fires. 

Last and importantly, In the very same fashion, I also have  a button component (functional ) and it works fine with `onClick` ( no use of native there ). Can anyone please let me know what's going on ?

**Comment by liqueflies** [comment]
Hi @kushalmahajan,
I opened a discussion about that at #67.

Please share your ideas :)

**Comment by kushalmahajan** [comment]
Please continue. I've given up on Vue already :)

**Comment by liqueflies** [comment]
Fixed with #65.
